src/webui/api/torrentscontroller.cpp

```diff
@@ -255,19 +264,23 @@ namespace
 void TorrentsController::infoAction()
 {
     const QString filter {params()["filter"]};
-    const QString category {params()["category"]};
-    const QString tag {params()["tag"]};
+    const std::optional<QString> category = getOptionalString(params(), QLatin1String("category"));
+    const std::optional<QString> tag = getOptionalString(params(), QLatin1String("tag"));
     const QString sortedColumn {params()["sort"]};
     const bool reverse {parseBool(params()["reverse"]).value_or(false)};
     int limit {params()["limit"].toInt()};
     int offset {params()["offset"].toInt()};
     const QStringList hashes {params()["hashes"].split('|', Qt::SkipEmptyParts)};
 
-    TorrentIDSet idSet;
-    for (const QString &hash : hashes)
-        idSet.insert(BitTorrent::TorrentID::fromString(hash));
+    std::optional<TorrentIDSet> idSet;
+    if (hashes.isEmpty())
```

Comment:
You probably meant the reverse test condition, right? Otherwise the enclosed for loop will never run.

Reply:
> You probably meant the reverse test condition, right? Otherwise the enclosed for loop will never run.

Sure. Fixed.